### PR TITLE
Add ability to ->forceJson() responses on requests

### PR DESCRIPTION
**Pull request by garygreen**
This PR allows you to call `$request->forceJson()` in your API to hint to Laravel you expect to send a JSON response where applicable.

The problem with the current approach is:

1. It's difficult to test API-based responses currently in the browser, unless you send content-type headers / use a tool like Postman, which maybe overkill for some to just test basic api
2. If your app only handles JSON-based responses and a non-json response header is sent, your app won't know how to respond with `ValidationException` etcs and will just endlessly redirect. With this PR, it will always send a JSON response of errors, no matter what the client requests.

**tldr;** creates more pleasant testing api's for developers, less exploitable by client requests.

### CODE REVIEW

**Comment by taylorotwell** [comment]
You probably want to make the check in `expectsJson` instead. That is the method that is used throughout the framework.

**Comment by deleugpn** [comment]
Instead of adding a new attribute and a check, wouldn't it be better if `forceJson()` just do something like `return $this->header('Accept, 'application/json')`?
By simply overriding the header, expectsJson will return true already.

**Comment by garygreen** [comment]
> Instead of adding a new attribute and a check, wouldn't it be better if `forceJson()` just do something like `return $this->header('Accept', 'application/json')`?

As far as I'm aware, you cannot modify the request headers in Symphony or anywhere in Laravel. If you could, I probably wouldn't of bothered with the PR as it's easy enough to add a json accept header and as you say, Laravel will take care of the rest.

@taylorotwell yes maybe just inside the `expectsJson` would be better as that function makes more "guesses" and assumptions about the request, whereas `wantsJson` is probably reserved for literally telling if the original response wants a json request. As long as `expectsJson` is used everywhere across the framework then shouldn't be a problem. Will update the PR

**Comment by garygreen** [comment]
PR has been updated so it's only checked in `expectsJson`

**Comment by deleugpn** [comment]
I have been using this for a couple of years now so I know it works. Ref: https://twitter.com/TheAlexLichter/status/969879256271597568

**Comment by garygreen** [comment]
Hmm interesting, that does seem to work and makes more sense than this PR. Thanks! 😄
